qa/rpc-tests/bip64.py

```diff
@@ -0,0 +1,130 @@
+        self.send_message(self.last_getheaders)
+
+        got_headers = wait_until(lambda: self.last_headers != None)
+        assert(got_headers)
+        self.send_message(self.last_headers)
+
+
+class BIP64_test(BitcoinTestFramework):
+
+    def __init__(self):
+        self.num_nodes = 2
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.xt_node = self.nodes[0]
+
+        # generate a block to get out of IBD
+        self.xt_node.generate(1)
+
+        self.test_node = TestNode()
+        self.test_node.add_connection(NodeConn('127.0.0.1', p2p_port(0), self.xt_node, self.test_node))
+
+        NetworkThread().start()
+        self.test_node.handshake()
+        self.getutxos_tests(self.xt_node)
+
+
+    def wait_for_utxo(self):
+        got_utxos = wait_until(lambda: self.test_node.utxos != None)
+        assert(got_utxos)
+        assert_equal(self.test_node.utxos.hash, int(self.xt_node.getbestblockhash(), 16))
+        assert_equal(self.test_node.utxos.height, self.xt_node.getblockcount())
+
+    def get_utxos(self, outpoints, checkmempool):
+        self.test_node.utxos = None
+        self.test_node.send_message(msg_getutxos(checkmempool, outpoints))
+        self.wait_for_utxo()
+
+    def getutxos_tests(self, xt_node):
+        txid = xt_node.sendtoaddress(xt_node.getnewaddress(), 0.1)
+        tx = FromHex(CTransaction(), xt_node.getrawtransaction(txid))
+        tx.rehash()
+
+        new_outpoint = COutPoint(tx.sha256, 0)
+        prev_outpoint = tx.vin[0].prevout
+
+        # Test that the utxo doesn't exist in the chain.
+        self.get_utxos([new_outpoint], checkmempool = False)
+        assert_equal(self.test_node.utxos.bitmap[0], int('0', 2))
+        assert_equal(len(self.test_node.utxos.result), 0)
+
+        # It does exist in the mempool.
+        self.get_utxos([new_outpoint], checkmempool = True)
+        assert_equal(self.test_node.utxos.bitmap[0], int('1', 2))
+        assert_equal(len(self.test_node.utxos.result), 1)
+        magic_inmempool_height = 2147483647
+        assert_equal(self.test_node.utxos.result[0].height, magic_inmempool_height)
+
+        # The prevout exists in the chain
+        self.get_utxos([prev_outpoint], checkmempool = False)
+        assert_equal(self.test_node.utxos.bitmap[0], int('1', 2))
+
+        # The prevout is spent in the mempool.
+        self.get_utxos([prev_outpoint], checkmempool = True)
+        assert_equal(self.test_node.utxos.bitmap[0], int('0', 2))
+
+        # Mine tx creating new_outpoint, now it should exist in the chain.
+        xt_node.generate(1)
+        self.get_utxos([new_outpoint], checkmempool = False)
+        assert_equal(self.test_node.utxos.bitmap[0], int('1', 2))
+        assert_equal(self.test_node.utxos.result[0].height, xt_node.getblockcount())
+
+        # .. same result when including mempool
+        self.get_utxos([new_outpoint], checkmempool = False)
```

Comment:
checkmempool = True here?
Testing that a bogus COutPoint appears in neither mode would also be nice.